### PR TITLE
DBZ-8291 DBZ-6709 Handle store.only.captured.tables.ddl regression

### PR DESCRIPTION
**Pull request by ani-sha**
closes https://issues.redhat.com/browse/DBZ-8291 and https://issues.redhat.com/browse/DBZ-6709

### CODE REVIEW

File: debezium-connector-binlog/src/test/java/io/debezium/connector/binlog/BinlogSnapshotSourceIT.java [hunk 1]
```diff
@@ -155,7 +154,7 @@ private void snapshotOfSingleDatabase(boolean useGlobalLock, boolean storeOnlyCa
                     .with(BinlogConnectorConfig.USER, "cloud")
```
Comment (ani-sha) [inline debezium-connector-binlog/src/test/java/io/debezium/connector/binlog/BinlogSnapshotSourceIT.java:154]: Modification in the test is required due to the conditional behavior of `useGlobalLock` here. Initially `STORE_ONLY_CAPTURED_TABLES_DDL` was set as true by default irrespective of the lock but now only this condition sets the config option.

**Comment by jpechane** [comment]
@ani-sha Rebased and force-pushed

**Comment by ani-sha** [comment]
Hi @mfvitale, I am uncertain about this failure. Do you believe it's connected to the change? Thanks.
```
2024-12-18T07:42:45.5123349Z [ERROR] NotificationsIT>AbstractNotificationsIT.notificationNotSentIfNoChannelIsConfigured:135->AbstractConnectorTest.assertNoRecordsToConsume:1029 expected:<[tru]e> but was:<[fals]e>
```

**Comment by jpechane** [comment]
@ani-sha Hi, yes, it is related. You need to run whole test `NotificationsIT` locally, test method in isolation won't show it. It seems that the test expects no messages but probably schema changes are sent.

**Comment by jpechane** [comment]
@ani-sha I tried to push a cumiit with the fix.

**Comment by ani-sha** [comment]
Thanks @jpechane, it is working!

**Pull request merged**